plugins/module_utils/network/sonic/config/l3_interfaces/l3_interfaces.py

```diff
@@ -164,16 +166,17 @@ def _state_overridden(self, want, have, diff):
         """
         ret_requests = list()
         commands = list()
-        interfaces_to_delete = get_diff(have, want, TEST_KEYS)
-        if interfaces_to_delete:
-            delete_interfaces_requests = self.get_delete_l3_interfaces_requests(want, have)
-            ret_requests.extend(delete_interfaces_requests)
-            commands.extend(update_states(interfaces_to_delete, "deleted"))
+        new_want = self.update_object(want)
+        l3_interfaces_to_delete = self.get_interface_object_for_overridden(have)
+        diff = get_diff(l3_interfaces_to_delete, new_want, TEST_KEYS)
 
         if diff:
-            interfaces_to_create_requests = self.get_create_l3_interfaces_requests(diff, have, want)
+            delete_interfaces_requests = self.get_delete_all_requests(diff)
```

Comment:
This will only delete configuration for interfaces specified in the executing playbook, but "overridden" state is defined as replacing all configuration for the affected resource module with only the configuration specified in the playbook.

For L3 interfaces, we could modify this to avoid deleting configuration for the management 0 interface, but all other interface configuration should be deleted if the playbook specifies a change for any interface.

Reply:
Updated the requested changes.
Now Mangement0 is exempted from delete handling.
Also, instead of diff, all l3 interfaces(except eth0) will be removed during overridden.